### PR TITLE
Add a 3d viewer

### PR DESCRIPTION
**Pull request by nikolas**
let me know if this works for you

I added a three.js animated scene triggered like a forttree theme:
screen shot 2015-03-18 at 11 40 34 pm

I also put `application.js` in a self-executing function so it doesn't pollute the global namespace.

### CODE REVIEW

File: app/assets/javascripts/application.js [hunk 1]
```diff
@@ -1,107 +1,121 @@
...
+    var my_text = $('#leaf_content').val();
+    var caret_pos = $('#leaf_content').getSelection().end;
 
-  $('#leaf_content').val(
-    my_text.substring(0, caret_pos)
-    + face
-    + my_text.substring(caret_pos)
-    );
+    $('#leaf_content').val(
+      my_text.substring(0, caret_pos)
```
Comment (nikolas) [inline app/assets/javascripts/application.js]: dude shut up

**Comment by mcfiredrill** [comment]
Wow amazing, I'll try this out. 

Sorry about Hound CI lol....

**Comment by mcfiredrill** [comment]
@nikolas Wow this is crazy ahaha, seems like not all of the images load for me, sometimes they are white, or looks like a loading spinner or something:

screenshot 2015-03-19 13 43 55
screenshot 2015-03-19 13 23 13

**Comment by nikolas** [comment]
yeah i had some loading problems too.. it seems random, i'll try to fix that.

**Comment by nikolas** [comment]
@mcfiredrill i got rid of the spinner :spaghetti:

**Comment by mcfiredrill** [comment]
Looks great now. 
screenshot 2015-04-01 07 58 24

**Comment by mcfiredrill** [comment]
Awesome

**Comment by mcfiredrill** [comment]
Can you rebase? @nikolas

**Comment by nikolas** [comment]
@mcfiredrill done.. but if you told @hound to buzz off that would help too

**Comment by mcfiredrill** [comment]
Thanks!

**Pull request merged**